tests/router_test.cc

```diff
@@ -248,6 +248,17 @@ TEST(router_test, test_route_head_request)
     endpoint->shutdown();
 }
 
+TEST(router_test, test_remove_not_existing)
+{
+    SegmentTreeNode routes;
+
+    ASSERT_THROW(routes.removeRoute("/v1/hello"), std::runtime_error);
+    ASSERT_THROW(routes.removeRoute("/v1/hello/:name/"), std::runtime_error);
+    ASSERT_THROW(routes.removeRoute("/get/:key?/bar"), std::runtime_error);
+    ASSERT_THROW(routes.removeRoute("/say/*/to/*"), std::runtime_error);
+    ASSERT_THROW(routes.removeRoute("*/api"), std::runtime_error);
```

Comment:
I'd prefer this to check for the exact exception that's expected to be thrown so that it is more obvious what this is testing, using `EXPECT_THAT` and `testing::ThrowsMessage`, like this (you also have to include the `gmock/gmock-matchers.h` header):

```suggestion
    using testing::ThrowsMessage;

    ASSERT_THAT(
        [&] { routes.removeRoute("/v1/hello"); },
        ThrowsMessage<std::runtime_error>("Requested route does not exist."));
    ASSERT_THAT(
        [&] { routes.removeRoute("/v1/hello/:name/"); },
        ThrowsMessage<std::runtime_error>("Requested route does not exist."));
    ASSERT_THAT(
        [&] { routes.removeRoute("/get/:key?/bar"); },
        ThrowsMessage<std::runtime_error>("Requested route does not exist."));
    ASSERT_THAT(
        [&] { routes.removeRoute("/say/*/to/*"); },
        ThrowsMessage<std::runtime_error>("Requested route does not exist."));
    ASSERT_THAT(
        [&] { routes.removeRoute("*/api"); },
        ThrowsMessage<std::runtime_error>("Requested route does not exist."));
```